test/acceptance/tests/basic/basic_test.go

```diff
@@ -352,217 +352,306 @@ func TestValidation_ConsulEcsConfigVariable(t *testing.T) {
 	}
 }
 
+// Test the validation that if a partition is provided, a namespace must also be provided.
+func TestValidation_NamespaceIsRequiredIfPartitionIsSet(t *testing.T) {
+	t.Parallel()
+
+	cases := map[string]struct {
+		partition string
+		namespace string
+		errMsg    string
+	}{
+		"without partition and namespace": {
+			partition: "",
+			namespace: "",
+			errMsg:    "",
+		},
+		"with partition and namespace": {
+			partition: "default",
+			namespace: "default",
+			errMsg:    "",
+		},
+		"with partition, without namespace": {
+			partition: "default",
+			namespace: "",
+			errMsg:    "ERROR: consul_namespace must be set if consul_partition is set",
+		},
+		"without partition, with namespace": {
+			partition: "",
+			namespace: "default",
+			errMsg:    "",
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
```

Comment:
non-blocking: For plan-only tests, I had found it's a touch faster to do the `terraform init` once outside the loop, and `t.Parallel()` here within the loop.

ex: https://github.com/hashicorp/terraform-aws-consul-ecs/blob/b8444236a962711e6070bd420723b2d07b99d8e6/test/acceptance/tests/basic/basic_test.go#L154-L157

Reply:
I've updated the test as suggested.